@xen-orchestra/web/src/components/pool/PoolHostInternalNetworkTable.vue

```diff
@@ -0,0 +1,301 @@
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import {
+  faAlignLeft,
+  faCaretDown,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPlus,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref, watch, watchEffect } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const props = defineProps<{
+  hostInternalNetwork: XoNetwork[]
+  isReady: boolean
+  hasError: boolean
+  selectedRowId: string | null
+}>()
+
+const emit = defineEmits<{
+  rowSelectHostInternalNetwork: [value: any]
+}>()
+
+const pagination = ref<PaginationPayload>({
+  currentPage: 1,
+  pageSize: 10,
+  startIndex: 0,
+  endIndex: 0,
+})
+
+const { t } = useI18n()
+const reactiveHostInternalNetworks = ref<XoNetwork[]>(props.hostInternalNetwork || [])
+
+const findPageById = () => {
+  const index = props.hostInternalNetwork.findIndex(network => network.id === props.selectedRowId)
+  if (index === -1) return null
+  pagination.value.currentPage = Math.floor(index / pagination.value.pageSize) + 1
+}
+
+watch(
+  () => props.hostInternalNetwork,
+  () => {
+    findPageById()
+  }
+)
+
+const selectRow = (item: any, table: string) => {
+  emit('rowSelectHostInternalNetwork', { item, table })
+}
+
+const searchQuery = ref('')
+
+const filteredNetworks = computed(() => {
+  let filtered = props.hostInternalNetwork
+  if (!searchQuery.value)
+    return props.hostInternalNetwork.slice(pagination.value.startIndex - 1, pagination.value.endIndex)
+  filtered = props.hostInternalNetwork.filter(network =>
+    Object.values(network).some(value => String(value).toLowerCase().includes(searchQuery.value.toLowerCase()))
+  )
+  return filtered.slice(pagination.value.startIndex - 1, pagination.value.endIndex)
+})
+
+const usableIds = computed(() => reactiveHostInternalNetworks.value.map(item => item.id))
+
+const { selected, areAllSelected } = useMultiSelect(usableIds)
+
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? usableIds.value : []
+}
+
+const { visibleColumns, rows } = useTable('networks', filteredNetworks, {
+  rowId: record => record.id,
+  columns: define => [
+    define('checkbox', () => '', { label: '', isHideable: false }),
+    define('name_label', { label: 'Name', isHideable: true }),
+    define('name_description', { label: 'Description', isHideable: true }),
+    define('MTU', { label: 'MTU', isHideable: true }),
+    define('defaultIsLocked', { label: 'Default Locking mode', isHideable: true }),
+    define('more', () => '', { label: '', isHideable: false }),
+  ],
+})
+
+type networkHeader = 'name_label' | 'name_description' | 'MTU' | 'defaultIsLocked' | 'more'
+
+const headerIcon: Record<networkHeader, { icon: IconDefinition }> = {
+  name_label: { icon: faAlignLeft },
+  name_description: { icon: faAlignLeft },
+  MTU: { icon: faHashtag },
+  defaultIsLocked: { icon: faCaretDown },
+  more: { icon: faEllipsis },
```

Comment:
No need to have an object with only one element, you can use the IconDefinition directly:
```suggestion
  name_label: faAlignLeft,
  name_description: faAlignLeft,
  MTU: faHashtag,
  defaultIsLocked: faCaretDown,
  more: faEllipsis,
```
Remember to update the rest of the code accordingly.